#' Functions
project.header <- function(aliasID,title,description,filename){
  write(paste(
    '<?xml version="1.0" encoding="UTF-8"?>',
    '<PROJECT_SET xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">',
    paste0('<PROJECT center_name="UPSC" alias="',aliasID,'">'),
    paste0("<TITLE>",title,"</TITLE>"),
    paste0("<DESCRIPTION>",description,"</DESCRIPTION>"),
    "<UMBRELLA_PROJECT/>",
    "<RELATED_PROJECTS>",
    sep="\n"),
    file=filename)
}

project.footer <- function(filename){
  write(paste(
    "</RELATED_PROJECTS>",
    "</PROJECT>",
    "</PROJECT_SET>",
    sep="\n"),
    file=filename,append = TRUE)
}

project.body <- function (pID,filename){
  write(paste(
    "<RELATED_PROJECT>",
    paste0('<CHILD_PROJECT accession="',pID,'"/>'),
    "</RELATED_PROJECT>",
    sep="\n"),
    file=filename,append = TRUE)
}

project.submission <- function(aliasID,projectFilename,holdDate,filename){
  write(
    paste('<?xml version="1.0" encoding="UTF-8"?>',
          '<SUBMISSION_SET xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">',
          paste0('<SUBMISSION alias="',aliasID,'" center_name="UPSC">'),
          "<ACTIONS>",
          "<ACTION>",
          paste0('<ADD source="',projectFilename,'" schema="project"/>'),
          "</ACTION>",
          "<ACTION>",
          paste0('<HOLD HoldUntilDate="',holdDate,'"/>'),
          "</ACTION>",
          "</ACTIONS>",
          "</SUBMISSION>",
          "</SUBMISSION_SET>",
          sep="\n"),
    file=filename)
}

echo.commands <- function(project.filename,receipt.filename,submission.filename){
  message( sprintf('curl -k -F "SUBMISSION=@%s" -F "PROJECT=@%s" "https://www-test.ebi.ac.uk/ena/submit/drop-box/submit/?auth=ENA%%20Webin-763%%20AdDfb2VJ" > %s',
                   submission.filename,project.filename,sub("\\.xml","-test.xml",receipt.filename)))
  message( sprintf('curl -F "SUBMISSION=@%s" -F "PROJECT=@%s" "https://www.ebi.ac.uk/ena/submit/drop-box/submit/?auth=ENA%%20Webin-763%%20AdDfb2VJ" > %s',
                   submission.filename,project.filename,receipt.filename))
}

# Call
# UPSC-0097
setwd("~/Git/UPSCb/projects/spruce-diversity/doc/")
alias="UPSC-0097"
project.filename=paste0(alias,".Project.xml")
submission.filename=paste0(alias,".Submission.xml")
title <- "Interspecific plastome recombination in Picea (Pinaceae) reveals ancient reticulate evolution"
description <- "Plastid sequences are a cornerstone in plant evolutionary inference and their interpretation is often guided by strong assumptions. Biparental inheritance, heteroplasmy, and sexual recombination appear rare, but detecting exceptions to these rules requires extensive taxon sampling and abundant sequence data. Utilizing advancements in high-throughput sequencing, we analyzed the whole plastomes of 65 accessions of Picea, a genus of approximately 35 coniferous forest trees, to test for violations of canonical plastome evolution. Using complementary hypothesis and data-driven tests, we found evidence for plastomes generated by interspecific hybridization and sexual recombination in the clade comprising Norway spruce (P. abies) and ten other species. Support for interspecific recombination remained after controlling for sequence saturation, positive selection, and potential alignment artifacts. These results resolve conflicting plastid-based phylogenies reported in previous studies and strengthen the inference of reticulate evolution in Picea. Given the relatively high frequency of biparental plastid inheritance and hybridization in plants, we suggest interspecific plastome recombination may be more widespread than currently appreciated and could underlie reported cases of discordant plastid phylogenies"

project.header(alias,title, description, project.filename)
dev.null <- sapply(
  read.csv("~/Box Sync/UPSC/Facility/Doc/Alexis/Studie-IDs.csv")[,2],
  function(pID){
  project.body(pID,project.filename)
})
project.footer(project.filename)

project.submission(alias,project.filename,Sys.Date()+729,submission.filename)

echo.commands(project.filename, submission.filename)

# UPSC-0100
setwd("~/Git/UPSCb/projects/spruce-pine-light/doc/ENA")
alias="UPSC-0100"
project.filename=paste0(alias,".Project.xml")
submission.filename=paste0(alias,".Submission.xml")
title <- "Effect of light quality in spruce and pine"
description <- "Scots pine and Norway Spruce are two main economically important tree species in Sweden. These two conifer species have contrasting characters with reference to response to shade: pine is shade intolerant while spruce is shade tolerant, therefore pine hypocotyls grow taller in shade as compared to sun while spruce hypocotyls show similar growth under shade and sun. Previous studies have described the ecotypes related to photoperiod and light spectra in forest trees and the involvement of light spectra in local adaptation has also been documented, yet molecular basis of response to shade is poorly understood so far. Transcriptome sequencing would allow identifying vital genes underlying the mechanism referring to the adaptation to the light quality, which would facilitate to comprehend whether the southern trees grown in the northern latitudes will be able to adapt to the light conditions prevailing in the north and gain biomass; or will they suffer due to light quality or light intensity and day length in the north. Thus this study will throw light on an area of prime importance relevant to the forest industry in Sweden."

project.header(alias,title, description, project.filename)
dev.null <- sapply(
  read.csv("~/Box Sync/UPSC/Facility/Doc/Sari/Studies-IDs.csv")[,2],
  function(pID){
    project.body(pID,project.filename)
  })
project.footer(project.filename)

project.submission(alias,project.filename,Sys.Date()+729,submission.filename)

echo.commands(project.filename, submission.filename)

# UPSC-0107
setwd("~/Git/UPSCb/projects/bellini_cambium/doc/ENA")
alias="UPSC-0107"
project.filename=paste0(alias,".Project.xml")
submission.filename=paste0(alias,".Submission.xml")
title <- "Transcriptome analysis of vascular cambium from easy-to-root and difficult-to-root genotype of populus"
description <- "Populus has considerable commercial and ecological importance and is accepted as a model system for biological studies of trees, and vegetative propagation of Populus plants chiefly depends on the formation of adventitious roots in stem cuttings. Adventitious roots (AR) are formed in stems, leaves and non-pericycle tissue in older roots, differing from primary roots, of embryonic origin. Developing adventitious root primordia (ARP) emerge from reprogrammed cells. Hence, the cellular origin of AR is uncertain and different tissues, often associated with the vasculature or cambial layers, which are proposed to generate AR founder cells. To better understand potential mechanisms involved in adventitious root formation in Populus, we performed a comparative transcriptome analysis at two time point (0 hour and 24 hours after cutting) between OP42 (P. maximowiczii  x P. trichocarpa) (easy-to-root) and hybrid aspen (P.tremula x P. tremuloides) clone T89 (difficult-to-root) from micro dissected cambium tissue using the Illumina sequencing method. Our data will be a valuable resource for genomic research about adventitious root formation in Populus cuttings, which can be used to improve rooting of difficult-to-root varieties."

project.header(alias,title, description, project.filename)
dev.null <- sapply(
  read.csv("~/Git/UPSCb/projects/bellini_cambium/doc/ENA/Studies-IDs.csv")[,2],
  function(pID){
    project.body(pID,project.filename)
  })
project.footer(project.filename)

project.submission(alias,project.filename,Sys.Date()+729,submission.filename)

echo.commands(project.filename, submission.filename)

# UPSC-0115
setwd("~/Git/UPSCb/projects/aspseq/doc/ENA")
alias="UPSC-0115"
project.filename=paste0(alias,".Project.xml")
submission.filename=paste0(alias,".Submission.xml")
title <- "GENOME AND POPULATION SEQUENCING OF ASPEN ADVANCES POPULUS AS A MODEL SYSTEM FOR PLANT RESEARCH"
description <- "Aspens include European trembling (Populus tremula) and north American quaking (P. tremuloides) aspen, European white aspen (P. alba) and numerous Asian species. Aspens spread as clonal colonies that can persist for millennia, are keystone species supporting a wide diversity of species and produce a diverse array of secondary metabolites. Here, we present genome and population genetics characterisation of P. tremula and P. tremuloides. Our analyses show that while the aspen genome structure is similar to that of the P. trichocarpa reference genome, with highly conserved gene content, they are considerably more heterozygous. One notable exception is the sex-determination region, which has altered gene content compared to P. trichocarpa. We used population re-sequencing data to examine the relationship between gene characteristics, genomic context and evidence of selection, finding that 5 prime UTRs have high rates of positive and negative selection and confirming a negative relationship between gene co-expression connectivity and purifying selection. The aspen genomics resources presented here greatly improve the utility of Populus as a model system for functional and evolutionary genomics. Data collections and analysis tools related to this study are available at PopGenIE (popgenie.org)."

project.header(alias,title, description, project.filename)
dev.null <- sapply(
  read.csv("~/Git/UPSCb/projects/aspseq/doc/ENA/Studies-IDs.csv")[,2],
  function(pID){
    project.body(pID,project.filename)
  })
project.footer(project.filename)

project.submission(alias,project.filename,Sys.Date()+729,submission.filename)

echo.commands(project.filename, submission.filename)

# UPSC-0120
setwd("~/Git/UPSCb/projects/arabidopsis-FD/doc/ENA")
alias="UPSC-0120"
project.filename=paste0(alias,".Project.xml")
submission.filename=paste0(alias,".Submission.xml")
title <- "Effects of FT on FD during the transition to flowering at the SAM in Arabidopsis thaliana"
description <- "The transition to flowering is a crucial step in the plant life cycle that is controlled by multiple endogenous and environmental cues, including hormones, sugars, temperature, and photoperiod. Permissive photoperiod induces FLOWERING LOCUS T (FT) in the phloem companion cells of leaves. The FT protein then acts as a florigen that is transported to the shoot apical meristem (SAM) where it physically interacts with the bZIP transcription factor FD and 14-3-3 proteins. However, despite the importance of FD for promoting flowering, its direct transcriptional targets are largely unknown. Here we applied ChIP-seq and RNA-seq to identify targets of FD at the genome scale and assess the contribution of FT to binding DNA. We further investigated the ability of FD to form protein complexes with FT and TFL1 through the interaction with 14-3-3 proteins. Importantly, we observe direct binding of FD to targets involved in the jasmonate signaling pathway, which so far had not been directly linked to the regulation of flowering time. Our study confirms FD as central regulator of the floral transition at the shoot meristem and provides evidence for crosstalk between the regulation of flowering and jasmonate signaling."

project.header(alias,title, description, project.filename)
dev.null <- sapply(
  read.csv("~/Git/UPSCb/projects/arabidopsis-FD/doc/ENA/Studies-IDs.csv")[,2],
  function(pID){
    project.body(pID,project.filename)
  })
project.footer(project.filename)

project.submission(alias,project.filename,Sys.Date()+729,submission.filename)

echo.commands(project.filename, submission.filename)

# UPSC-0124
setwd("~/Git/UPSCb/projects/spruce-needles/doc/ENA")
alias="UPSC-0124"
project.filename=paste0(alias,".Project.xml")
submission.filename=paste0(alias,".Submission.xml")
title <- "Overwintering the boreal winter: From seasonal gene expression and photosynthetic responses to acclimation of conifers"
description <- "The northern boreal forest is dominated by evergreen conifers. The needles of the most abundant species of the European boreal forest (Pinus sylvestrys and Picea abies) are able to survive the extremely cold (freezing) temperatures during the long dark winter and fully recover during the spring. Conifer needles possess potent regulatory processes that protect its photosynthetic apparatus during the harsh boreal winter (very low temperatures and very low light) and spring (low temperature and high light). However, the precise molecular and/or physiological mechanisms of this plasticity are not well understood. In this manuscript we discuss the seasonal adaptation of photosynthetic machineries of Scots pine (Pinus sylvestris) and Norway spruce (Picea abies) and suggest possible mechanisms allowing them to survive the harsh Scandinavian winter. First, we measure seasonal changes of gene expression (using RNA-Seq) of the major photosynthetic components in Norway spruce. Secondly, we use chlorophyll fluorescence to profile photosynthetic activities during two consecutive winters in needles of both Norway spruce and Scots pine. The functional performance of photosynthetic electron transport (Photosystem II and Photosystem I photochemistry) indicate the highly dynamic structural and functional seasonal rearrangement of the photosynthetic apparatus involved in balancing the electron flow through linear and PSI-dependent cyclic electron transfer pathways during the seasonal transitions. The involvement of a number of alternative electron transport pathways including the PTOX- and flv-dependent electron sinks during the overwintering period and recovery of photosynthesis in the spring is also established."

project.header(alias,title, description, project.filename)
dev.null <- sapply(
  read.csv("Studies-IDs.csv")[,2],
  function(pID){
    project.body(pID,project.filename)
  })
project.footer(project.filename)

project.submission(alias,project.filename,Sys.Date()+729,submission.filename)

echo.commands(project.filename, submission.filename)

# UPSC-0127
setwd("~/Git/UPSCb/projects/metagenomics/doc/ENA")
alias="UPSC-0127"
project.filename=paste0(alias,".Project.xml")
submission.filename=paste0(alias,".Submission.xml")
title <- "Comparing reproducibility and biological information using RNA Sequencing metatranscriptomics and DNA amplicon metagenomics"
description <- "This is a placeholder for the abstract"

project.header(alias,title, description, project.filename)
dev.null <- sapply(
  read.csv("Studies-IDs.csv")[,2],
  function(pID){
    project.body(pID,project.filename)
  })
project.footer(project.filename)

project.submission(alias,project.filename,Sys.Date()+729,submission.filename)

echo.commands(project.filename, submission.filename)

# UPSC-0133
setwd("~/Git/UPSCb/projects/Arabidopsis_RNA_cold_acclimation/doc/ENA")
alias="UPSC-0133"
project.filename=paste0(alias,".Project.xml")
submission.filename=paste0(alias,".Submission.xml")
title <- "New strategies to face long exposure to low temperatures in Norway spruce revealed by transcriptomic analysis"
description <- "Most cold acclimation gene expression studies have addressed short-term responses in herbaceous plant leaves so far but few have addressed the acclimation responses further 24 hours to freezing or of roots, and even fewer have investigated perennial evergreens such as conifers, which face extremely low temperatures during winter. 
To understand the transcriptional changes in needles during acclimation to low temperatures we performed genome-wide RNA-Seq experiments with non-dormant two-year old Norway spruce seedlings under long day conditions. In a time series experiment set up to mimic stepwise cold acclimation to low temperatures, plants were shifted from growth permissive (18 degrees Celsius) to cold (5 degrees Celsius) temperatures for up to ten days and subsequently to freezing temperatures (-5 degrees Celsius) for further ten days. For direct comparison with the standard herbaceous model, Arabidopsis thaliana leaves exposed to cold (5 degrees Celsius) were sampled at equivalent time points. Finally, Spruce root samples were collected from both chilling and freezing conditions for comparison with needle responses to assess whether the root cold response is initiated via the same regulatory pathways as in aboveground tissues.
"

project.header(alias,title, description, project.filename)
dev.null <- sapply(
  read.csv("Studies-IDs.csv")[,2],
  function(pID){
    project.body(pID,project.filename)
  })
project.footer(project.filename)

project.submission(alias,project.filename,Sys.Date()+729,submission.filename)

echo.commands(project.filename, submission.filename)

# UPSC-0115 - update
setwd("~/Git/UPSCb/projects/aspseq/doc/ENA")
alias="UPSC-0115"
project.filename=paste0(alias,"-update.Project.xml")
submission.filename=paste0(alias,"-update.Submission.xml")
title <- "GENOME AND POPULATION SEQUENCING OF ASPEN ADVANCES POPULUS AS A MODEL SYSTEM FOR PLANT RESEARCH"
description <- "Aspens include European trembling (Populus tremula) and north American quaking (P. tremuloides) aspen, European white aspen (P. alba) and numerous Asian species. Aspens spread as clonal colonies that can persist for millennia, are keystone species supporting a wide diversity of species and produce a diverse array of secondary metabolites. Here, we present genome and population genetics characterisation of P. tremula and P. tremuloides. Our analyses show that while the aspen genome structure is similar to that of the P. trichocarpa reference genome, with highly conserved gene content, they are considerably more heterozygous. One notable exception is the sex-determination region, which has altered gene content compared to P. trichocarpa. We used population re-sequencing data to examine the relationship between gene characteristics, genomic context and evidence of selection, finding that 5 prime UTRs have high rates of positive and negative selection and confirming a negative relationship between gene co-expression connectivity and purifying selection. The aspen genomics resources presented here greatly improve the utility of Populus as a model system for functional and evolutionary genomics. Data collections and analysis tools related to this study are available at PopGenIE (popgenie.org)."

project.header(alias,title, description, project.filename)
dev.null <- sapply(
  read.csv("~/Git/UPSCb/projects/aspseq/doc/ENA/Studies-IDs-update.csv")[,2],
  function(pID){
    project.body(pID,project.filename)
  })
project.footer(project.filename)

project.submission(alias,project.filename,Sys.Date()+729,submission.filename)

echo.commands(project.filename, submission.filename)

# UPSC-0139
setwd("~/Git/UPSCb/projects/arabidopsis-serrate-ChIP-Seq/doc/ENA")
alias="UPSC-0139"
project.filename=paste0(alias,".Project.xml")
submission.filename=paste0(alias,".Submission.xml")
title <- "TERMINAL FLOWER 1 functions as a mobile transcriptional co-factor in the Arabidopsis thaliana shoot apical meristem"
description <- "The floral transition is a critical step in the plant life cycle and several mechanisms control this finely orchestrated process. TERMINAL FLOWER 1 (TFL1) is a floral repressor and close relative of the florigen, FLOWERING LOCUS T (FT). During floral transition TFL1 expression is up-regulated in the apex to maintain the indeterminate growth of the shoot apical meristem (SAM). Both TFL1 and FT are mobile proteins, but they move in different ways. FT moves from the leaves to the SAM while TFL1 appears to move within the SAM. The importance of TFL1 movement for its function in the regulation of flowering time and shoot indeterminacy and its molecular function are still largely unclear. Our results indicate that the TFL1 protein moves from its place of expression in the centre of the SAM all the way to the meristem layer L1 where it regulates the floral transition. ChIP-seq and RNA-seq demonstrate that TFL1 functions as a co-transcription factor that associates with and regulates the expression of hundreds of genes. Some putative direct targets of TFL1 have been identified, opening the possibility to discover new roles for TFL1 in the regulation of floral transition and inflorescence development."

project.header(alias,title, description, project.filename)
dev.null <- sapply(
  read.csv("Studies-IDs.csv")[,2],
  function(pID){
    project.body(pID,project.filename)
  })
project.footer(project.filename)

project.submission(alias,project.filename,Sys.Date()+729,submission.filename)

echo.commands(project.filename, submission.filename)

# UPSC-0142
setwd("~/Git/UPSCb/projects/ERF/doc/ENA")
alias="UPSC-0142"
project.filename=paste0(alias,".Project.xml")
submission.filename=paste0(alias,".Submission.xml")
title <- "An AP/ERF transcription factor ERF139 coordinates xylem expansion and lignin deposition"
description <- "Wood formation is a result of massive deposition of secondary cell wall material, a process  tightly regulated by the complex interplay of  transcription factors. APETALA2/ETHYLENE RESPONSE FACTORs (AP/ERFs) are transcription factors that are often implicated in signaling of ethylene but that have not been investigated in wood formation even though ethylene is clearly important in this process. The function of  Populus trichocarpa ERF139 (Potri.013G101100) in wood formation was characterised in overexpression and dominant repression lines of ERF139 created by fusion of ERF139 with the EAR repression domain (SRDX). Pyrolysis-GC/MS analysis revealed opposite phenotypes in the two different types of transgenic trees, supporting the role of ERF139 in altering the flux of lignin biosynthesis towards the guaiacyl (G) type lignin and suppressing it towards the p-hydroxy phenyl (H) type lignin. The results further supported the role of ERF139 in suppressing the radial expansion of the secondary xylem vessel elements. RNA-Seq based comparative transcriptomics identified genes homologous to master regulators of secondary cell wall deposition (LBD15, ANAC2, MYB86) and genes related to lignin biochemistry (LACCASE5 and UDP-glycosyltransferase 72B1) as potential direct targets of ERF139. These results suggest that ERF139 is a master regulator of transcription factors that form part of regulatory cascades related to  xylem expansion and lignification."

project.header(alias,title, description, project.filename)
dev.null <- sapply(
  read.csv("Studies-IDs.csv")[,2],
  function(pID){
    project.body(pID,project.filename)
  })
project.footer(project.filename)

project.submission(alias,project.filename,Sys.Date()+729,submission.filename)

echo.commands(project.filename, submission.filename)

# UPSC-0149
setwd("~/Git/UPSCb/projects/arabidopsis-rn4-wgs/doc/ENA")
alias="UPSC-0150"
project.filename=paste0(alias,".Project.xml")
submission.filename=paste0(alias,".Submission.xml")
title <- "Selective auxin agonists induce specific AUX/IAA protein degradation to modulate plant development"
description <- "Auxin phytohormones control most aspects of plant development through a complex and interconnected signaling network. In the presence of auxin, AUXIN/INDOLE-3-ACETIC ACID (AUX/IAA) transcriptional repressors are targeted for degradation by the SKP1-CULLIN1-F-BOX (SCF) ubiquitin-protein ligases containing TRANSPORT INHIBITOR RESISTANT 1/AUXIN SIGNALING F-BOX (TIR1/AFB). CULLIN1-neddylation is required for SCFTIR1/AFB functionality as exemplified by mutants deficient in the NEDD8-activating enzyme subunit AUXIN-RESISTANT 1 (AXR1). Here, we report a chemical biology screen that identifies small molecules requiring AXR1 to modulate plant development. We selected four molecules of interest, RubNeddin1 to 4 (RN1 to 4), among which RN3 and RN4 trigger selective auxin responses at transcriptional, biochemical and morphological levels. This selective activity is explained by their ability to consistently promote the interaction between TIR1 and a specific subset of AUX/IAA proteins, stimulating the degradation of particular AUX/IAA combinations. Finally, we performed a genetic screen using RN4, the RN with the greatest potential for dissecting auxin perception, which revealed that the chromatin remodeling ATPase BRAHMA is implicated in auxin-mediated apical hook development. These results demonstrate the power of selective auxin agonists to dissect auxin perception for plant developmental functions as well as offering opportunities to discover new molecular players involved in auxin responses."

project.header(alias,title, description, project.filename)
dev.null <- sapply(
  read.csv("Studies-IDs.csv")[,2],
  function(pID){
    project.body(pID,project.filename)
  })
project.footer(project.filename)

project.submission(alias,project.filename,Sys.Date()+729,submission.filename)

echo.commands(project.filename, submission.filename)

# UPSC-0148
setwd("~/Git/UPSCb/projects/leaf_development-srna/doc/ENA")
alias="UPSC-0148"
project.filename=paste0(alias,".Project.xml")
submission.filename=paste0(alias,".Submission.xml")
receipt.filename=paste0(alias,".Receipt.xml")
title <- "Complex genetic architecture of leaf shape is associated with gene expression variation in Populus tremula"
description <- "Leaf shape is a defining feature of how we recognise and classify plant species. Although there is extensive variation in leaf shape within many species, few studies have disentangled the underlying genetic architecture. Identification of the genetic components controlling variation in such complex traits would enable future applications such as re-engineering developmental programs to maximise growth performance under different abiotic conditions. We characterised the genetic architecture of leaf shape variation in Eurasian aspen (Populus tremula L.) by performing a genome wide association study (GWAS). To ascertain whether identified GWAS candidates were central within the leaf development transcriptional program, we performed gene co-expression network analyses from developmental series profiling contrasting leaf forms. We additionally performed regression analyses to identify developmental gene expression profiles predictive of physiognomy traits. We identified genes that were differentially expressed between groups of genotypes with contrasting leaf shapes from the GWAS population and examined their developmental expression profiles. These data were integrated with expression GWAS results to define a set of candidate gene associated with leaf shape variation. Our results indicate that leaf shape traits are genetically complex, likely determined by numerous small-effect variations in gene expression. Genes associated with shape variation were peripheral within the population-wide co-expression network, were not central regulators within the leaf development gene expression program and exhibited signatures of relaxed selection. Genes identified included numerous novel candidates with no known function in leaf development. These results provide targets for functional characterisation and a comprehensive resource for exploring gene expression during leaf development in aspen, available at http://aspleaf.plantgenie.org."
project.header(alias,title, description, project.filename)
dev.null <- sapply(
  read.csv("Studies-IDs.csv")[,2],
  function(pID){
    project.body(pID,project.filename)
  })
project.footer(project.filename)

project.submission(alias,project.filename,Sys.Date()+729,submission.filename)

echo.commands(project.filename, receipt.filename, submission.filename)

